### ISSUE TITLE
XSLT warning "The parent axis starting at a document node will never select anything"

### ISSUE THREAD

**Issue by danielweck** [w3c/epubcheck#859]
Hello!

* I am on MacOS (latest)
* I installed the Oracle JDK (latest v11.0.1 from https://www.oracle.com/technetwork/java/javase/downloads/jdk11-downloads-5066655.html )
* I installed Maven (latest)

* I cloned this repository and checkout'ed the `master` branch
* I invoked the Maven install command (no problems)
* I ran EpubCheck from the command line against a few EPUBs (seems to work just fine).
* However, the following message is reported over 20 times in the console (just before the actual validation message):

```
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
```

```
Validating using EPUB version 3.0.1 rules.
No errors or warnings detected.
Messages: 0 fatal / 0 errors / 0 warnings / 0 info
```

System:

`java --version`
```
java 11 2018-09-25
Java(TM) SE Runtime Environment 18.9 (build 11+28)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11+28, mixed mode)
```

`mvn --version`
```
Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T19:33:14+01:00)
Maven home: /Applications/apache-maven
Java version: 11, vendor: Oracle Corporation, runtime: /Library/Java/JavaVirtualMachines/jdk-11.jdk/Contents/Home
Default locale: en_GB, platform encoding: UTF-8
OS name: "mac os x", version: "10.14", arch: "x86_64", family: "mac"
```

`java -jar target/epubcheck.jar --version`
```
EpubCheck v4.0.3-SNAPSHOT
No file specified in the arguments. Exiting.
epubcheck completed
```

**Comment by rdeltour** [comment]
> Warning at char 3 in xsl:param/@select on line 459 column 43 SXWN9000: The parent axis starting at a document node will never select anything

Would there be a file associated somewhere in the log with this line and column number?
If not we'll have to do some spelunking :-)

**Comment by tofi86** [comment]
Note that there are no Travis checks for Java 11 yet. We should set those up and check if it’s a Java 11 issue.

**Comment by danielweck** [comment]
I've been looking "everywhere" (EPUBCheck wiki docs, CLI help, known MacOS tmp / app folders, GitHub issue tracker) ... but I am not able to find the full EPUBCheck log. Where should I look for it? Thanks! :)

**Comment by danielweck** [comment]
I also checked the list of system-level file handles opened/closed by the Java process (MacOS / Darwin), I cannot see any low-level log file. Perhaps there isn't one? In which case, all I have is the log excerpt quoted in my original post (above).

**Comment by rdeltour** [comment]
> I've been looking "everywhere" (EPUBCheck wiki docs, CLI help, known MacOS tmp / app folders, GitHub issue tracker) ... but I am not able to find the full EPUBCheck log. Where should I look for it?

Sorry I wasn't clear: there's no log file, I was asking if there was more context in the console output from your previous message. Apparently not 😄

**Comment by tofi86** [comment]
> Note that there are no Travis checks for Java 11 yet. We should set those up and check if it’s a Java 11 issue.

I added a travis check for JDK 11 with PR #860 but it does not throw any errors like these. So it seems to be an issue with your special book.

**Comment by GrayWolfMT** [comment]
I'm not sure if is related but I'm on a Windows 10 system with the latest version of the source code, java version 1.8 and Maven 3.5.4 and I'm getting a similar behavior.  I just did a pull from the repo and did get an update to the travis.yaml file, but no other changes.

When I clean and package the code, I get the exact same message that Daniel is getting when I clean and package the code.  It is generated by the "com.adobe.epubcheck.api.Epub20CheckExpandedTest" test suite, and even though there are only 22 tests in the set, I see the error 27 times.

I have tried this in both the "standard" Windows command prompt and using the Git Bash shell as well; both produce the exact same behavior.

I'm just getting started looking at the code base here, and I wasn't sure if this is an expected behavior for that test, but in light of the issue, I wanted to include the information.

java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)

mvn -version
Apache Maven 3.5.4 (1edded0938998edf8bf061f1ceb3cfdeccf443fe; 2018-06-17T12:33:14-06:00)
Maven home: C:\Program Files (x86)\Apache\maven\bin\..
Java version: 1.8.0_181, vendor: Oracle Corporation, runtime: C:\Program Files\Java\jdk1.8.0_181\jre
Default locale: en_US, platform encoding: Cp1252
OS name: "windows 10", version: "10.0", arch: "amd64", family: "windows"

**Comment by danielweck** [comment]
`mvn install` runs the tests "default-test" and "coverage-test" (all pass), respectively:

```
Tests run: 1002, Failures: 0, Errors: 0, Skipped: 1
```
and:
```
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
```

Note that "default-test" reports the warning message:
```
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
```

Full console log:

```
[INFO] --- maven-surefire-plugin:2.18.1:test (default-test) @ epubcheck ---
[INFO] Surefire report directory: /Users/danielweck/Code/epubcheck/target/surefire-reports

-------------------------------------------------------
 T E S T S
-------------------------------------------------------
Running com.adobe.epubcheck.api.Epub20CheckExpandedTest
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 422 column 41 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 422 column 41 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Tests run: 22, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 14.626 sec - in com.adobe.epubcheck.api.Epub20CheckExpandedTest
Running com.adobe.epubcheck.api.Epub20CheckTest
Spine element not found within package root document ops/manifest.opf
Spine element not found within package root document EPUB/manifest.opf
Spine element not found within package root document OEBPS/content.opf
Tests run: 33, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 5.681 sec - in com.adobe.epubcheck.api.Epub20CheckTest
Running com.adobe.epubcheck.api.Epub30CheckExpandedTest
Tests run: 148, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 16.55 sec - in com.adobe.epubcheck.api.Epub30CheckExpandedTest
Running com.adobe.epubcheck.api.Epub30CheckTest
Tests run: 48, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 13.984 sec - in com.adobe.epubcheck.api.Epub30CheckTest
Running com.adobe.epubcheck.cli.CLITest
Tests run: 21, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.01 sec - in com.adobe.epubcheck.cli.CLITest
Running com.adobe.epubcheck.nav.NavCheckerTest
Tests run: 13, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.175 sec - in com.adobe.epubcheck.nav.NavCheckerTest
Running com.adobe.epubcheck.ocf.OCFCheckerTest
Tests run: 24, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.836 sec - in com.adobe.epubcheck.ocf.OCFCheckerTest
Running com.adobe.epubcheck.ocf.OCFFilenameCheckerTest
Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.018 sec - in com.adobe.epubcheck.ocf.OCFFilenameCheckerTest
Running com.adobe.epubcheck.opf.MetadataSetTest
Tests run: 13, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.075 sec - in com.adobe.epubcheck.opf.MetadataSetTest
Running com.adobe.epubcheck.opf.OPFCheckerTest
Tests run: 128, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.884 sec - in com.adobe.epubcheck.opf.OPFCheckerTest
Running com.adobe.epubcheck.ops.OPSCheckerTest
Tests run: 100, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.485 sec - in com.adobe.epubcheck.ops.OPSCheckerTest
Running com.adobe.epubcheck.overlay.OverlayCheckerTest
Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.183 sec - in com.adobe.epubcheck.overlay.OverlayCheckerTest
Running com.adobe.epubcheck.test.api_Test
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.297 sec - in com.adobe.epubcheck.test.api_Test
Running com.adobe.epubcheck.test.command_line_Test
Tests run: 19, Failures: 0, Errors: 0, Skipped: 1, Time elapsed: 2.809 sec - in com.adobe.epubcheck.test.command_line_Test
Running com.adobe.epubcheck.test.css_Test
Tests run: 17, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 4.543 sec - in com.adobe.epubcheck.test.css_Test
Running com.adobe.epubcheck.test.dtBook_Test
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.438 sec - in com.adobe.epubcheck.test.dtBook_Test
Running com.adobe.epubcheck.test.encryption_Test
Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.919 sec - in com.adobe.epubcheck.test.encryption_Test
Running com.adobe.epubcheck.test.opf_Test
Tests run: 30, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 6.114 sec - in com.adobe.epubcheck.test.opf_Test
Running com.adobe.epubcheck.test.package_Test
Tests run: 22, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.006 sec - in com.adobe.epubcheck.test.package_Test
Running com.adobe.epubcheck.test.script_Test
Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.655 sec - in com.adobe.epubcheck.test.script_Test
Running com.adobe.epubcheck.test.single_file_Test
Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.1 sec - in com.adobe.epubcheck.test.single_file_Test
Running com.adobe.epubcheck.test.toc_Test
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.256 sec - in com.adobe.epubcheck.test.toc_Test
Running com.adobe.epubcheck.test.xhtml_Test
Tests run: 17, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.649 sec - in com.adobe.epubcheck.test.xhtml_Test
Running com.adobe.epubcheck.util.OPFPeekerTest
Tests run: 19, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.092 sec - in com.adobe.epubcheck.util.OPFPeekerTest
Running com.adobe.epubcheck.util.PathUtilTest
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0 sec - in com.adobe.epubcheck.util.PathUtilTest
Running com.adobe.epubcheck.vocab.PrefixParsingTest
Tests run: 13, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.068 sec - in com.adobe.epubcheck.vocab.PrefixParsingTest
Running com.adobe.epubcheck.vocab.PropertyTest
Tests run: 3, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 sec - in com.adobe.epubcheck.vocab.PropertyTest
Running com.adobe.epubcheck.vocab.VocabTest
Tests run: 14, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.076 sec - in com.adobe.epubcheck.vocab.VocabTest
Running org.idpf.epubcheck.util.css.CssInputStreamTest
Tests run: 11, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.007 sec - in org.idpf.epubcheck.util.css.CssInputStreamTest
Running org.idpf.epubcheck.util.css.CssParserTest
Tests run: 121, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.229 sec - in org.idpf.epubcheck.util.css.CssParserTest
Running org.idpf.epubcheck.util.css.CssScannerTest
Tests run: 127, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.032 sec - in org.idpf.epubcheck.util.css.CssScannerTest

Results :

Tests run: 1002, Failures: 0, Errors: 0, Skipped: 1
```

**Comment by danielweck** [comment]
With `childrens-literature.epub` downloaded from http://idpf.github.io/epub3-samples/30/samples.html

`java -jar target/epubcheck.jar childrens-literature.epub `

=>

```
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 422 column 41 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 422 column 41 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Validating using EPUB version 3.0.1 rules.
No errors or warnings detected.
Messages: 0 fatal / 0 errors / 0 warnings / 0 info

epubcheck completed
```

**Comment by danielweck** [comment]
With `accessible_epub_3.epub` downloaded from http://idpf.github.io/epub3-samples/30/samples.html

`java -jar target/epubcheck.jar accessible_epub_3.epub`

=>

```
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 422 column 41 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 422 column 41 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
Validating using EPUB version 3.0.1 rules.
ERROR(CSS-020): /Users/danielweck/Downloads/accessible_epub_3.epub/EPUB/css/epub.css(130,3): CSS font selector declaration uses unexpected font-size value 'bold'.

Check finished with errors
Messages: 0 fatal / 1 errors / 0 warnings / 0 info

epubcheck completed
```

**Comment by danielweck** [comment]
So, instead of Oracle JDK v11.0.1 from https://www.oracle.com/technetwork/java/javase/downloads/jdk11-downloads-5066655.html , I have tried with OpenJDK v11.0.1 from https://jdk.java.net/11/

`/usr/libexec/java_home -V`
```
Matching Java Virtual Machines (2):
    11.0.1, x86_64:	"OpenJDK 11.0.1"	/Library/Java/JavaVirtualMachines/jdk-11.0.1.jdk/Contents/Home
    11, x86_64:	"Java SE 11"	/Library/Java/JavaVirtualMachines/jdk-11.jdk/Contents/Home

/Library/Java/JavaVirtualMachines/jdk-11.0.1.jdk/Contents/Home
```

`/usr/libexec/java_home -v 11`
```
/Library/Java/JavaVirtualMachines/jdk-11.0.1.jdk/Contents/Home
```

`java --version`
```
openjdk 11.0.1 2018-10-16
OpenJDK Runtime Environment 18.9 (build 11.0.1+13)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.1+13, mixed mode)
```

I get the exact same console messages (as detailed above), so I won't repeat them here. Excerpt:
```
Warning at char 3 in xsl:param/@select on line 459 column 43 
  SXWN9000: The parent axis starting at a document node will never select anything
```

**Comment by rdeltour** [comment]
I had a fresh look at this issue and I can reproduce it locally too (on macOS 10.13.6 running OpenJDK 11.0.1+13).

The issue comes from [the schematron XSLT bundled with Jing](https://github.com/relaxng/jing-trang/blob/V20151127/mod/schematron/src/main/com/thaiopensource/validate/schematron/resources/iso-schematron.xsl#L459), which has the `/..` XPath expression which will always evaluate to the empty sequence.

Apparently Saxon 9.8 raises a `SXWN9000` warning for this construct, when Saxon 9.7 stayed silent. I ran the tests before #830 was applied and the warning didn't show up.

It puzzles me however why the warning doesn't happen on the Travis build… to be further investigated.

As for making this warning disappear, we have several options:
- revert to Saxon 9.7
- try to configure the Saxon instance run by Jing to stay silent
- try to override the XSLT bundled with Jing by a patched XSLT where we remove this selector

I will also file a bug to Jing.
